File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        start,end = 0, len(arr)
        while start < end:
            mid = start + (end - start) // 2
            if(arr[mid] > arr[mid+1]):
                end = mid 
            else:
                start = mid + 1
        return end

File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int start(0), end(A.size());
        while(start < end)
        {
            int mid = start + (end - start) / 2;
            if(A[mid] > A[mid+1]) end = mid ;
            else start = mid + 1;
        }
        return end;
    }
};